### ISSUE TITLE
New map update causes screen to blank out on initial load

### ISSUE THREAD

**Issue by mixart** [FlutterFlow/flutterflow-issues#2796]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When navigating to a new page with the map widget, the entire screen blanks out while the widget loads (takes a few seconds). This is new since the latest map update and can be seen on web and mobile. This only happens the first time the map loads, but makes it feel like the app crashed since all navigation on top/bottom of the app disappears.

### Expected Behavior

Expect the core components like nav, etc. to not be refreshed in the UI and remain on screen.

I also noticed some warnings in the browser console that talk about not loading async (not sure if this is related.

<img width="1046" alt="image" src="https://github.com/FlutterFlow/flutterflow-issues/assets/13832/a33862fb-1d05-43c8-901f-a40a5a8a1e64">

### Steps to Reproduce

1. Create a page with some navigation on it
2. Create a second page with a map widget
3. navigate from page 1 to page 2
4. the app will blank out for several seconds
5. Note that the more markers, the longer it takes to load the widget, the longer the screen is blank.

Project to repro here:
https://app.flutterflow.io/project/map-test-6seeb6

https://github.com/FlutterFlow/flutterflow-issues/assets/13832/753a2375-7424-44e9-8b81-468de6d14368

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFfi/LqyItIsdtK0L6IK8AwmjwsG1McTL4ZlO0bRQsjGuv2PZh+PvfRbRZvTc/iT2lhHFe0nlEz7vDviuzlJ/EDPQeZGqJ++ZF+aj3vd3q8SraLBoeeT3F6EfpPBX6kyrSngx54JulYL1Eu2FOQeufKUgHAJ5+/Zwh9f6PHaOI=

### Context

Cannot update my app since it has map functionality and app feels broken now.
Blocking release.

### Visual documentation

https://github.com/FlutterFlow/flutterflow-issues/assets/13832/4deccc15-db42-4456-b480-16d1693017ba

### Environment

```markdown
- FlutterFlow version: 4.1.34+
- Platform: web, mobile
- Browser name and version: Chrome
- Operating system and version affected:
```

### General

- [ ] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [ ] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [ ] ~100% of the time
- [ ] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by ignalauret** [comment]
Hey @mixart thanks for your report. Our technical team is checking if any recent update in the Markers could be causing this. Thanks for noticing this performance issue. They will keep you updated on this thread, have a great day!

**Comment by glgh** [comment]
@mixart thanks for reporting the bug. Unfortunately I can't seem to reproduce the screen blankout with your test project on either web or mobile - can you please confirm the issue still exists on the latest FF release?

**Comment by mixart** [comment]
@glgh i will try update the project with more data. The issue is more amplified when more markers are added but if want to try yourself, add 40 markers to a map and the blackout will appear longer.

**Comment by mixart** [comment]
Another user also reported the issue in the community here: https://community.flutterflow.io/c/whats-new-in-flutterflow/post/what-s-new-in-flutterflow-april-18-2024-xiQAr0TKHHL8oCO?highlight=Si7MrP9hyktmS9z

**Comment by glgh** [comment]
@mixart yes please, it will help us debug if you can add more markers to the project to exaggerate the lag.

Does the lag only occurs when the map center is set to current device location, or does it happen regardless?

**Comment by Maazkhalid47** [comment]
I am facing the same issue when navigating to a page that contains a map, now I have tested this in different ways, I have found that its not the map rather using the current device location action to bind the map's initial center to the user's current location that displays that screen with a loader while getting the device location
